force-app/main/default/classes/SOQL.cls

```diff
@@ -288,20 +427,36 @@ public with sharing class SOQL {
      */
     public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
+        List<Lead> leads = [SELECT Id, Email FROM Lead  WHERE Id IN :leadIds];
 
         // Create a set of email domains from the Leads
+        Set<String> emailDomains = new Set<String>();
 
         // Loop through each Lead
+            for (Lead lead : leads){
+                String email = lead.Email;
+                
+                // Check if the email is not null
+                if (email != null && email.contains('@')){
+                    String domain = email .split ('@')[1];
+                    emailDomains.add(domain);
+                }
+            }
 
             // Get the email from the Lead
 
-            // Check if the email is not null
-
                 // Add the email domain to the set
-
+                Set<String> websiteDomains = new Set<String>();
+                for (String Domain : emailDomains){
```

Comment:
Why have a second loop here? Couldn't you do this within the first loop you did that captured email domains?